website/docs/best-practices/custom-generic-tests.md

```diff
@@ -100,6 +100,30 @@ models:
 
 With one line of code, you've just created a test! In this example, `users` will be passed to the `is_even` test as the `model` argument, and `favorite_number` will be passed in as the `column_name` argument. You could add the same line for other columns, other models—each will add a new test to your project, _using the same generic test definition_.
 
+### Add description to generic data test logic
+
+You can add a description to the Jinja macro that provides the core logic for a data test by including the `description` key under the `macros:` section. You can add descriptions directly to the macro, including descriptions for macro arguments.
+
+Here's an example:
+
+```yaml
+macros:
+  - name: test_not_empty_string
+    description: Complementary test to default `not_null` test as it checks that there is not an empty string. It only accepts columns of type string. It is deprecated as this functionality has been included in the custom implementation of `not_null`.
```

Comment:
The last sentence was specific to the user that originally provided this example in https://github.com/dbt-labs/dbt-core/issues/2578#issuecomment-1868473901. So we should peel it out for our documentation purposes.

```suggestion
    description: Complementary test to default `not_null` test as it checks that there is not an empty string. It only accepts columns of type string.
```